boutdata/restart.py

```diff
@@ -1195,3 +1195,120 @@ def get_block(data):
             # Write fields
             for k in interp_data:
                 f.write(k, get_block(interp_data[k]))
+
+
+def shift_v3_to_v4(
+    gridfile, zperiod, path="data", output=".", informat="nc", mxg=2, myg=2
+):
+    """Convert a set of restart files from BOUT++ v3 to v4
+
+       Assumes that both simulations are using shifted metric coordinates:
+       - v3 restarts are in field-aligned coordinates
+         (shifted when taking X derivatives)
+       - v4 restarts are in shifted coordinates
+         (shifted when taking Y derivatives)
+
+    Parameters
+    ----------
+
+    gridfile : str
+        String containing grid file name
+    zperiod : int
+        Number of times the domain is repeated to form a full torus
+    path : str, optional
+        Directory containing the input restart files (BOUT++ v3)
+    output : str, optional
+        Directory where the output restart files will go
+    informat : str, optional
+        File extension of the input restart files
+    mxg : int, optional
+        Number of X guard cells
+    myg : int, optional
+        Number of Y guard cells
+    """
+
+    # Try opening the grid file
+    with DataFile(gridfile) as grid:
+        try:
+            zShift = grid["zShift"]
+        except KeyError:
+            zShift = grid["qinty"]
+
+    if path == output:
+        raise ValueError("Can't overwrite restart file")
+
+    file_list = glob.glob(os.path.join(path, "BOUT.restart.*." + informat))
+    file_list.sort()
```

Comment:
This will fail if there are more than 10 files, because `.sort()` thinks `BOUT.restart.10.nc` comes before `BOUT.restart.1.nc`. Need to use `natsort` - I'll push a fix...